java-checks/src/main/java/org/sonar/java/filters/LombokFilter.java

```diff
@@ -93,56 +99,44 @@
+    String fullyQualifiedName = ExpressionsHelper.concatenate((ExpressionTree) tree.qualifiedIdentifier());
 
-    if (generatesEquals || usesAnnotation(tree, GENERATE_UNUSED_FIELD_RELATED_METHODS)) {
-      excludeLines(tree, UnusedPrivateFieldCheck.class);
-      excludeLines(tree, PrivateFieldUsedLocallyCheck.class);
-    } else {
-      acceptLines(tree, UnusedPrivateFieldCheck.class);
-      acceptLines(tree, PrivateFieldUsedLocallyCheck.class);
-    }
+    excludeLinesIfTrue(fullyQualifiedName.startsWith("lombok."), tree, UselessImportCheck.class);
 
-    if (usesAnnotation(tree, GENERATE_CONSTRUCTOR)) {
-      excludeLines(tree, AtLeastOneConstructorCheck.class);
-    } else {
-      acceptLines(tree, AtLeastOneConstructorCheck.class);
-    }
+    super.visitImport(tree);
+  }
 
-    if (generatesEquals) {
-      excludeLines(tree, EqualsNotOverriddenInSubclassCheck.class);
-      excludeLines(tree, EqualsNotOverridenWithCompareToCheck.class);
-    } else {
-      acceptLines(tree, EqualsNotOverriddenInSubclassCheck.class);
-      acceptLines(tree, EqualsNotOverridenWithCompareToCheck.class);
-    }
+  @Override
+  public void visitClass(ClassTree tree) {
+    boolean generatesEquals = usesAnnotation(tree, GENERATE_EQUALS);
 
-    if (generatesPrivateConstructor(tree)) {
-      excludeLines(tree, UtilityClassWithPublicConstructorCheck.class);
-    } else {
-      acceptLines(tree, UtilityClassWithPublicConstructorCheck.class);
-    }
+    excludeLinesIfTrue(generatesEquals || usesAnnotation(tree, GENERATE_UNUSED_FIELD_RELATED_METHODS), tree, UnusedPrivateFieldCheck.class, PrivateFieldUsedLocallyCheck.class);
+    excludeLinesIfTrue(usesAnnotation(tree, GENERATE_CONSTRUCTOR), tree, AtLeastOneConstructorCheck.class);
+    excludeLinesIfTrue(generatesEquals, tree, EqualsNotOverriddenInSubclassCheck.class, EqualsNotOverridenWithCompareToCheck.class);
+    excludeLinesIfTrue(generatesPrivateConstructor(tree), tree, UtilityClassWithPublicConstructorCheck.class);
+    excludeLinesIfTrue(usesAnnotation(tree, UTILITY_CLASS), tree, BadFieldNameCheck.class, ConstantsShouldBeStaticFinalCheck.class);
+
+    super.visitClass(tree);
+  }
 
-    if (usesAnnotation(tree, UTILITY_CLASS)) {
-      excludeLines(tree, BadFieldNameCheck.class);
-      excludeLines(tree, ConstantsShouldBeStaticFinalCheck.class);
+  private final void excludeLinesIfTrue(boolean shouldExclude, Tree tree, Class<? extends JavaCheck> rule) {
+    if (shouldExclude) {
+      excludeLines(tree, rule);
     } else {
-      acceptLines(tree, BadFieldNameCheck.class);
-      acceptLines(tree, ConstantsShouldBeStaticFinalCheck.class);
+      acceptLines(tree, rule);
     }
+  }
 
-    super.visitClass(tree);
+  @SafeVarargs
+  private final void excludeLinesIfTrue(boolean shouldExclude, Tree tree, Class<? extends JavaCheck>... rules) {
+    BiConsumer<Tree, Class<? extends JavaCheck>> consumer = shouldExclude ? this::excludeLines : this::acceptLines;
```

Comment:
I find it a bit strange to declare lambda variable, which is used on the next line. I don't think using Stream API gives a lot in terms of clarity here, I would just use old fashioned
```
for (Class<? extends JavaCheck> rule : rules) {
      if (shouldExclude) {
        excludeLines(tree, rule);
      } else {
        acceptLines(tree, rule);
      }
    }```

Reply:
I wanted to avoid to repeat the boolean test for the potentially N checks from the varargs. But as there is only max 2 of them at the moment, it's maybe overkill.